### DISCUSSION TITLE
Indirect Vulkan Rendering

### DISCUSSION THREAD

**Discussion by JJ-IsShort** [AdaptiveCpp/AdaptiveCpp#1304] [Q&A]
I was just wondering if it was possible to use AdaptiveCpp to calculate a bunch of values on the GPU, then use those in a shader, without moving the kernel result off of the GPU? Vulkan can do indirect rendering so I was hoping it was possible to convert AdaptiveCpp buffers to Vulkan buffers somehow.

**Comment by illuhad** [comment]
This is possible for some backends, but not universally since there's no universal model for how compute APIs can talk to graphics APIs, and because AdaptiveCpp also supports non-GPU devices like CPUs. Generally, this can be done if there is interoperability with Vulkan for your SYCL backend via backend-specific mechanisms.

For example, CUDA supports Vulkan interoperability, and AdaptiveCpp guarantees direct interoperability of SYCL device pointers with CUDA device pointers if you run through the CUDA backend. So you can use those SYCL device pointers to use CUDA-Vulkan interoperability. 

On the AMD side, Vulkan interopability in the ROCm compute stack is there, but I don't know how mature it is.
The OpenCL and Level Zero backends currently do not support interoperability.
